crates/uv-resolver/src/lock/mod.rs

```diff
@@ -625,6 +625,11 @@ impl Lock {
         })
     }
 
+    /// Returns the package whose name matches [`PackageName`].
+    pub fn find_package(&self, name: &PackageName) -> Option<&Package> {
```

Comment:
We could make it `find_packages` instead to iterate once and retrieve the packages we are interested in.